src/emmy/expression/compile.cljc

```diff
@@ -499,31 +300,51 @@
   [f-form]
   (sci/eval-form (sci/fork sci-context) f-form))
 
-(defn- compile-state->source
-  "Returns a natively-evaluated Clojure function that implements `body`, given:
+(defn- compile->clj
+  "Returns Clojure source for a function that implements `body`, given:
 
   - `params`: a seq of symbols equal in count to the original state function's
     args
   - `state-model`: a sequence of variables representing the structure of the
     nested function returned by the state function
   - `body`: a function body making use of any symbol in the args above"
-  [params state-model body opts]
+  [argv body]
   (let [body (w/postwalk-replace sym->resolved-form body)]
-    `(fn ~(state-argv params state-model opts) ~body)))
+    `(fn [~@argv] ~body)))
+
+(defn- compile->js
+  "Returns an array containing JavaScript source for a function that implements
+  `body` in a form suitable for the arguments of the Function constructor, given:
+
+  - `params`: a seq of symbols equal in count to the original state function's
+    args
+  - `state-model`: a sequence of variables representing the structure of the
+    nested function returned by the state function
+  - `body`: a function body making use of any symbol in the args above"
+  [argv body]
+  (let [argv        (mapv commafy-arglist argv)
+        js          (render/->JavaScript* body)
+        cs-bindings (s/join (for [[var val] (:vars js)]
+                              (str "  const " var " = " val ";\n")))
```

Comment:
is there any benefit to using `let` here, or is it the other way around, that `var` has higher performance but potentially less safety?

Reply:
The [experts say](https://stackoverflow.com/questions/40070631/v8-javascript-performance-implications-of-const-let-and-var) it doesn't matter. I'm on the fence. Having been raised on const-correctness in ANSI C, I like to use const when possible, but if you write JS in modern style, practically everything is const except loop indices.